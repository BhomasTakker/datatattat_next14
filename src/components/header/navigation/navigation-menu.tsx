import { NavLinkData } from "./nav-link";
import { NavigationLinks } from "./navigation-links";
import classes from "./navigation-menu.module.scss";

interface NavigationMenuProps {
	items: NavLinkData[];
	edit?: boolean;
	prefix?: string;
}

export const NavigationMenu = ({ items, prefix = "" }: NavigationMenuProps) => {
	const showNav = items && items.length > 0;

	return (
		<>
			{showNav ? (
				// div seems unnecessary
				<div className={classes.root}>
					<nav className={classes.toolbar} data-testid={"nav"}>
						<NavigationLinks navLinks={items} prefix={prefix} />
					</nav>
				</div>
			) : null}
		</>
	);
};
