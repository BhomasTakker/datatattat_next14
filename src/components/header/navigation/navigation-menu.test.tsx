import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import { NavigationMenu } from "./navigation-menu";

jest.mock("./navigation-links", () => {
	return {
		NavigationLinks: ({
			navLinks,
		}: {
			navLinks: { route: string; label: string }[];
		}) => {
			return (
				<div>
					{navLinks.map(({ label, route }) => (
						<a key={label} data-testid={label} href={route}>
							{label}
						</a>
					))}
				</div>
			);
		},
	};
});

const MOCK = [
	{
		route: "/home",
		label: "Home",
	},
	{
		route: "/about",
		label: "About",
	},
];

describe("Navigation Menu", () => {
	it("renders nav element when provided links", () => {
		render(<NavigationMenu items={MOCK} />);
		const nav = screen.getByTestId("nav");
		expect(nav).toBeInTheDocument();
	});

	it("renders provided links", () => {
		render(<NavigationMenu items={MOCK} />);
		const link = screen.getByRole("link", { name: /home/i });
		expect(link).toBeInTheDocument();
	});

	it("does not render a nav element when provided with an empty array", () => {
		render(<NavigationMenu items={[]} />);
		const nav = screen.queryByTestId("nav");
		expect(nav).not.toBeInTheDocument();
	});

	it("renders navigation unchanged", () => {
		const { container } = render(<NavigationMenu items={MOCK} />);
		expect(container).toMatchSnapshot();
	});
});
