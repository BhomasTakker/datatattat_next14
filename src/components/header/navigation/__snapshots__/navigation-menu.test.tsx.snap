// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Menu renders navigation unchanged 1`] = `
<div>
  <div
    class="root"
  >
    <nav
      class="toolbar"
      data-testid="nav"
    >
      <div>
        <a
          data-testid="Home"
          href="/home"
        >
          Home
        </a>
        <a
          data-testid="About"
          href="/about"
        >
          About
        </a>
      </div>
    </nav>
  </div>
</div>
`;
